### PR TITLE
distutils-r1.eclass: next patch batch

### PR DESCRIPTION
**Pull request by mgorny**
Collecting patches.

CC @gentoo/python

### CODE REVIEW

File: eclass/distutils-r1.eclass
Comment (AndrewAmmerlaan) [inline eclass/distutils-r1.eclass:207]: This I think is missing a `:`:
```
: ${DISTUTILS_ALLOW_WHEEL_REUSE:=1}
```

File: eclass/distutils-r1.eclass
Comment (eli-schwartz) [inline eclass/distutils-r1.eclass:1410]: This is an inferior version of my patch, without my signoff. :(

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: eclass/distutils-r1.eclass
Comment (eli-schwartz) [reply eclass/distutils-r1.eclass:207]: Both with and without a `:` are valid bash and mean two different things.

- with the `:` -- if `DISTUTILS_ALLOW_WHEEL_REUSE=` is set and blanked in the environment, it falls back to 1
- without the `:` -- if `DISTUTILS_ALLOW_WHEEL_REUSE=` is set and blanked in the environment, it is still *set* so we do nothing, and simply respect the value in the environment.

Comment (eli-schwartz) [reply eclass/distutils-r1.eclass:207]: Since later on we check whether the value is non-empty, it is utterly pointless to add `:` because then any attempt to override it to empty would trigger falling back to the eclass value. Supporting "set it in the environment to a blank value" is an intentional design choice.

File: eclass/distutils-r1.eclass
Comment (eli-schwartz) [reply eclass/distutils-r1.eclass:1410]: 8d0f2de98faf0a7ac82dd9b1b465550722647163

In particular, the wheel name is an obscure way to describe it -- it's an output filename, and it's an issue because the process has nominally succeeded with a successful return code.

Comment (mgorny) [reply eclass/distutils-r1.eclass:1410]: I have been waiting for your reply to the comment on your patch for a month or so. I don't really care. "Internal error" makes little sense to me, it's build system misbehaving.

Comment (mgorny) [reply eclass/distutils-r1.eclass]: Ok, reverted the change. I don't need your accusations.

Comment (thesamesam) [reply eclass/distutils-r1.eclass]: It's a provenance thing, it's not an accusation - you obviously did it because of the other patch, so mention it.

But really, don't leave comments on commits, nobody looks at them?

Comment (mgorny) [reply eclass/distutils-r1.eclass]: So perhaps you should submit changes correctly instead of throwing links to detached commits on IRC.

**Comment by chewi** [comment]
The changes I'd like to get in soon are from [my cargo-env branch](https://github.com/gentoo/gentoo/compare/master...chewi:gentoo:cargo-env). The Python changes depend on the Cargo changes, so they'd all need to go in with anything you're doing here.

**Comment by mgorny** [comment]
> The changes I'd like to get in soon are from [my cargo-env branch](https://github.com/gentoo/gentoo/compare/master...chewi:gentoo:cargo-env). The Python changes depend on the Cargo changes, so they'd all need to go in with anything you're doing here.

Is it ok if I merge them in here, and then mark the whole thing ready for review?

**Comment by mgorny** [comment]
I've merged @chewi 's changes in, and I believe this is now ready for review.

**Review by chewi — APPROVED** [review]

**Review by AndrewAmmerlaan — APPROVED** [review]

**Pull request closed without merging**
